packages/transport/src/api/udp.ts

```diff
@@ -33,13 +34,31 @@ export class UdpApi extends AbstractApi {
             if (!this.listening) return;
 
             this.enumerationTimeout = setTimeout(() => {
-                this.enumerate(this.enumerateAbortController.signal).finally(enumerateRecursive);
+                this.enumerate(this.enumerateAbortController.signal)
+                    .then(r => this.handleEnumerate(r))
+                    .finally(enumerateRecursive);
             }, 500);
         };
 
         enumerateRecursive();
     }
 
+    private handleEnumerate(result: Awaited<ReturnType<typeof this.enumerate>>) {
```

Comment:
enumerate in usb assings it's result to to the this.devices variable. https://github.com/trezor/trezor-suite/blob/develop/packages/transport/src/api/usb.ts#L176
 
maybe this fn should be called also somewhere on line 194?

Reply:
not sure if this is correct behavior, usb emits `transport-interface-change` from the `listen` function relying on the fact that `enumerate` was called before that. but what if it wasnt? what if i called listen without calling enumerate first?

btw, usb also assigns `this.devices` on line 59 and 82

Reply:
on second thought, if i go with expected way (first enumerate then listen) then `transport-interface-change` will be emitted regardless, so i agree that i should prevent that, but i cant assign `this.devices` (inside enumerate fn) before i check for changes (after enumerate)

i will think about it